File: FiscalFlow/FiscalFlow.Contracts/Transactions/UpdateTransaction.cs
using FiscalFlow.Domain.Enums;

namespace FiscalFlow.Contracts.Transactions;

public class UpdateTransaction
{
    public Guid TransactionId { get; set; }
    public string Description { get; set; }
    public string Payee { get; set; }
    public TransactionType Type { get; set; }
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }
    public decimal Value { get; set; }
    public ushort? Recurrence { get; set; }
    public Category Category { get; set; }
    public DateTime CreatedOnUtc { get; set; }
}

File: FiscalFlow/fiscalflow.client/src/app/shared/models/transaction/updateTransaction.ts
import { TransactionType } from './enums/transactionType';
import { Category } from './enums/category';

export interface UpdateTransaction {
  transactionId: string;
  description: string;
  payee: string;
  type: TransactionType;
  value: number;
  longitude: number | null;
  latitude: number | null;
  imageUrl: string | null;
  category: Category;
  createdOnUtc: Date;
}
